### PR TITLE
Make underlyingErrorMessage an empty string if inexistent in iOS

### PR DESCRIPTION
**Pull request by vegaro**
Just saw this issue I opened more than a year ago and couldn't believed I hadn't fixed before https://github.com/RevenueCat/purchases-hybrid-common/issues/3

This could cause issues in platforms where the map object that represents the error is directly converted to the hybrid SDK object, like in Cordova, where the error map is converted automatically to a JavaScript object. In the case of Cordova iOS, `underlyingErrorMessage` would be missing from the JavaScript object in iOS if there's no `underlyingErrorMessage` in the SKError.

For consistency, I made `underlyingErrorMessage` an empty string if it's missing in the SKError

### CODE REVIEW

**Review by aboedo — APPROVED** [review]
looks good. Ideally we'd move the method to a separate class and add unit tests for it, but I think we can do that as a follow-up if needed

**Comment by vegaro** [comment]
I will extract it, no worries. I will keep this open until I find a little bit of time to do it

**Comment by aboedo** [comment]
@vegaro what's the status on this one?

**Comment by vegaro** [comment]
I am going to extract to a separate class when I finish the etags stuff. We don't need to merge this right now so it can wait.

File: ios/PurchasesHybridCommon/PurchasesHybridCommon/RCCommonFunctionality.m [hunk 1]
```diff
@@ -310,21 +317,20 @@ + (void)paymentDiscountForProductIdentifier:(NSString *)productIdentifier
                 SKProductDiscount *discountToUse = [self discountWithIdentifier:discountIdentifier forProduct:aProduct];
 
                 if (discountToUse) {
+                    void (^paymentDiscountCompletion)(SKPaymentDiscount *, NSError *) =
+                    ^(SKPaymentDiscount *paymentDiscount, NSError *error) {
```
Comment (aboedo) [inline ios/PurchasesHybridCommon/PurchasesHybridCommon/RCCommonFunctionality.m:321]: don't you _love_ block syntax?

File: ios/PurchasesHybridCommon/PurchasesHybridCommon/RCErrorContainer.h [hunk 2]
```diff
@@ -5,13 +5,17 @@
 
 #import <Purchases/Purchases.h>
 
+NS_ASSUME_NONNULL_BEGIN
+
```
Comment (aboedo) [inline ios/PurchasesHybridCommon/PurchasesHybridCommon/RCErrorContainer.h:9]: 👏

File: ios/PurchasesHybridCommon/PurchasesHybridCommon/RCErrorContainer.h [hunk 3]
```diff
@@ -5,13 +5,17 @@
...
 @interface RCErrorContainer : NSObject
 
 @property (nonatomic, readonly) NSInteger code;
 @property (nonatomic, nonnull, readonly) NSString *message;
 @property (nonatomic, nonnull, readonly) NSDictionary *info;
 @property (nonatomic, nonnull, readonly) NSError *error;
```
Comment (aboedo) [inline ios/PurchasesHybridCommon/PurchasesHybridCommon/RCErrorContainer.h:15]: these no longer need `nonnull` since it's assumed

**Review by aboedo — APPROVED** [review]
looks great!

**Pull request merged**